internal/pkg/envconfig/envconfig.go

```diff
@@ -8,17 +8,24 @@
 package envconfig
 
 import (
-	"fmt"
 	"os"
 )
 
+// TODO:
+//
+// Vars like 'SINGULARITY_*' should be declared as const here
+//
+
+// NoCache will return true if 'SINGULARITY_DISABLE_CACHE' is set.
 func NoCache() bool {
 	env := os.Getenv("SINGULARITY_NO_CACHE")
-
-	fmt.Println("DEBIG: ENV: ", env)
-
-	if env == "1" || env == "true" {
+	if env != "" {
```

Comment:
```
$ export SINGULARITY_NO_CACHE=""
$ env | grep 'NO_CACHE'
SINGULARITY_NO_CACHE=
```

Being set like:  `export SINGULARITY_NO_CACHE=""` ... makes the variable set ... but an empty string. `os.LookupEnv` should probably be used in this case to differentiate between an empty value, and unset variable.